runtime/ops/worker_host.rs

```diff
@@ -131,361 +120,442 @@ pub fn init(create_web_worker_cb: Arc<CreateWebWorkerCb>) -> Extension {
     .build()
 }
 
-fn merge_boolean_permission(
-  mut main: UnitPermission,
-  worker: Option<PermissionState>,
-) -> Result<UnitPermission, AnyError> {
-  if let Some(worker) = worker {
-    if worker < main.state {
-      return Err(custom_error(
-        "PermissionDenied",
-        "Can't escalate parent thread permissions",
-      ));
-    } else {
-      main.state = worker;
-    }
-  }
-  Ok(main)
+fn escalation_error() -> AnyError {
+  custom_error(
+    "PermissionDenied",
+    "Can't escalate parent thread permissions",
+  )
```

Comment:
I'm wondering if these changes should mostly live in `runtime/permissions.rs`... Seems like serialization of permissions could be better suited there instead of `ops/worker_host.rs`. Thoughts?

Reply:
Also seems like `create_worker_permissions` could use some unit tests - should be easier (and faster) that having heavy integration tests